PayPalCheckout/src/main/java/com/paypal/android/checkout/PayPalLifeCycleObserver.kt

```diff
@@ -0,0 +1,12 @@
+package com.paypal.android.checkout
+
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+
+class PayPalLifeCycleObserver(private val payPalClient: PayPalClient) : LifecycleObserver {
+    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
+    fun resume() {
+        payPalClient.handleBrowserSwitchResult()
```

Comment:
I haven't tested this but apparently, the lifecycle gets notified after `onCreate()` returns so if we can get merchants to call `setListener()` immediately after instantiation within onCreate we should be good.